### PR TITLE
Add SSL Cert to foundation and use it for CloudFront

### PR DESCRIPTION
**Pull request by ialford**

### CODE REVIEW

File: cdk.context.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
       "useVpcId": "vpc-797fed1d",
       "createDns": true,
       "domainName": "libraries.nd.edu",
```
Comment (mlkirkpatrick) [inline cdk.context.json:8]: Is domainName still needed? Wouldn't you import this from the domainStack's exports?

**Review by mlkirkpatrick — APPROVED** [review]

**Review by rdoughty — APPROVED** [review]

**Review by SteveMattison — APPROVED** [review]

File: src/foundation-stack.ts [hunk 2]
```diff
@@ -1,16 +1,21 @@
 import * as cdk from '@aws-cdk/core'
 import { Bucket, BucketAccessControl } from '@aws-cdk/aws-s3'
+import { Certificate, ICertificate } from '@aws-cdk/aws-certificatemanager'
 
-export interface FoundationStackProps extends cdk.StackProps {}
+export interface FoundationStackProps extends cdk.StackProps {
+  readonly domainStack: string;
```
Comment (jgondron) [inline src/foundation-stack.ts:6]: Would rather this be called `domainStackName`. `domainStack` gives the impression its type is a Stack, meaning it's created by this app.

**Review by jgondron — CHANGES_REQUESTED** [review]

File: cdk.context.json (hunk 1 above)
Comment (jgondron) [reply cdk.context.json:8]: Second this

Comment (ialford) [reply cdk.context.json:8]: Should I just get rid of this context value then, and we can import as needed within the stacks?

Comment (ialford) [reply cdk.context.json:8]: @jgondron and @mlkirkpatrick - see my question above. Thoughts?

Comment (mlkirkpatrick) [reply cdk.context.json:8]: Yes, I think so.

File: src/foundation-stack.ts (hunk 2 above)
Comment (ialford) [reply src/foundation-stack.ts]: Updated

**Review by jgondron — APPROVED** [review]

**Pull request merged**